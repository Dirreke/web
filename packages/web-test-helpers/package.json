{
  "name": "web-test-helpers",
  "private": true,
  "peerDependencies": {
      "@vue/composition-api": "1.7.1",
      "vue-gettext": "^2.1.5",
      "vue-router": "3.6.5",
      "vuex": "^3.6.2"
  }
}
