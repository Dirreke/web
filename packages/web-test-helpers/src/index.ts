export * from './mocks/defaultComponentMocks'
export * from './mocks/defaultStubs'
export * from './mocks/axios'

export * from './defaultPlugins'
export * from './helpers'
