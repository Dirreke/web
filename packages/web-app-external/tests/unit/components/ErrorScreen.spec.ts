import ErrorScreen from '../../../src/components/ErrorScreen.vue'
import { mount } from 'web-test-helpers'

describe('The external app error screen component', () => {
  test('displays an icon and a paragraph', () => {
    const wrapper = mount(ErrorScreen, {
      props: {
        message: 'Error when loading the application'
      },
      global: {
        stubs: {
          OcIcon: true
        }
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
