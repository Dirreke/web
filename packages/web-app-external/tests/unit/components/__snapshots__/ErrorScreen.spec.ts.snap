// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The external app error screen component displays an icon and a paragraph 1`] = `
<div class="oc-text-center">
  <ocicon-stub name="error-warning" filltype="line" accessiblelabel="" type="span" size="xxlarge" variation="passive" color=""></ocicon-stub>
  <p class="oc-text-xlarge">Error when loading the application</p>
</div>
`;
